main.js

```diff
@@ -0,0 +1,319 @@
+       }
+   }
+   return youngestPerson;
+}
+
+function getOldestPerson(people) {
+    let oldestPerson = people[0].age;
+
+    for (i = 1; i < people.length; i++) {
+        if (people[i].age > oldestPerson) {
+            return oldestPerson = people[i].age;
+        }
+    }
+    return oldestPerson;
+}
+
+function getAgeDifference(people) {
+    const oldestPersonAge = getOldestPerson(peopleArray);
+    const youngestPersonAge = getYoungestPerson(peopleArray);
+
+    return oldestPersonAge - youngestPersonAge;
+}
+
+console.log(getAgeDifference(peopleArray));
+
+// 8 - codewars
+
+function getPositiveNumbersSum(numbersArray) {
+    let sum = 0;
+
+    for (i = 0; i < numbersArray.length; i++) {
+        let currentNumber = numbersArray[i];
+
+        if (currentNumber > 0) {
+            sum = sum + currentNumber;
+        }
+    }
+    return sum;
+}
+
+console.log(getPositiveNumbersSum([1,-2,3,4,5]));
+
+// 9
+
+function getSumOfSquaredNumbers(numbersArray){
+    let sum = 0;
+
+    for (i = 0; i < numbersArray.length; i++) {
+        let currentnumber = numbersArray[i];
+        sum = sum + (currentnumber * currentnumber);
+    }
+    return sum;
+}
+
+console.log(getSumOfSquaredNumbers([1, 2, 3]));
+
+// 10
+
+function getNameInitials(name){
+    const nameConvertedToArray = name.toUpperCase().split(' ');
+    return `${nameConvertedToArray[0][0]}.${nameConvertedToArray[1][0]}`;
+}
+
+console.log(getNameInitials('Sam Harris'));
+console.log(getNameInitials('Anita Włodarczyk'));
+
+// 11
+
+function countSheepsIfPresent(sheepArray) {
+    let sumOfSheepsPresent = 0;
+
+    for (i = 0; i < sheepArray.length; i++) {
+        let isCurrentSheepPresent = sheepArray[i];
+
+        if (isCurrentSheepPresent === true) {
+            sumOfSheepsPresent = sumOfSheepsPresent + 1;
+        }
+    }
+    return sumOfSheepsPresent;
+}
+
+console.log(countSheepsIfPresent([undefined,null,false,true,true,false,null,undefined]));
+
+// 12
+
+function getDigitizedAndReversedNumber(number) {
+    const numberSplit = number.toString().split('');
+    return numberSplit.reverse().map(Number);
+}
+
+console.log(getDigitizedAndReversedNumber(35231));
+
+// 13
+
+function findNeedleIndex(haystackArray) {
+    return "found the needle at position " + haystackArray.indexOf("needle");
+}
+
+console.log(findNeedleIndex(['283497238987234', 'a dog', 'a cat', 'some random junk', 'a piece of hay', 'needle', 'something somebody lost a while ago']));
+
+// 14
+
+function getDoubledNumbers(numberArray) {
+    function doubledNumbers(number) {
+        return number * 2;
+    }
+    return numberArray.map(doubledNumbers);
+}
+
+console.log(getDoubledNumbers([1, 2, 3]));
+
+// 15
+
+function getInvertedNumbersArray(numbersArray) {
+    return numbersArray.map(function(number) {
+        return number * -1;
+    });
+}
+
+console.log(getInvertedNumbersArray([1, -2, 3, -4, 5]));
+
+// 16
+
+function calculateNumbersInArraySum(numbersArray) {
+    "use strict";
```

Comment:
????

Reply:
It was as already inside the function in codewars. So i left it in.